src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -353,19 +362,26 @@ public Snapshot retrieveSnapshotByName(String name) {
       isolation = Isolation.SERIALIZABLE,
       readOnly = true)
   Snapshot retrieveWorker(String sql, MapSqlParameterSource params) {
+    RowMapper<Snapshot> mapper =
+        (rs, rowNum) -> {
+          try {
+            return new Snapshot()
+                .id(rs.getObject("id", UUID.class))
+                .name(rs.getString("name"))
+                .description(rs.getString("description"))
+                .createdDate(rs.getTimestamp("created_date").toInstant())
+                .profileId(rs.getObject("profile_id", UUID.class))
+                .projectResourceId(rs.getObject("project_resource_id", UUID.class))
+                .mode(
+                    objectMapper.readValue(
```

Comment:
What happens on cases where this is null?  We may want an explicit method that maps to the `SnapshotRequestContentsModel` (also has the added benefit of not requiring moving the entire mapper here into a try...catch block)

Reply:
It's probably worth pulling into a separate deserialization methods for when the SnapshotRequestContentsModel class evolves